src/code/z_collision_check.c

```diff
@@ -1,392 +1,3373 @@
 #include <ultra64.h>
 #include <global.h>
+#include <vt.h>
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005B280.s")
+void func_8005B2AC(GraphicsContext* gfx, Vec3f* vA, Vec3f* vB, Vec3f* vC, u8 r, u8 g, u8 b);
 
+// draw red poly
+void func_8005B280(GraphicsContext* gfx, Vec3f* vA, Vec3f* vB, Vec3f* vC) {
+    func_8005B2AC(gfx, vA, vB, vC, 255, 0, 0);
+}
+
+// draw poly
 #pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005B2AC.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005B65C.s")
+s32 Collider_InitBase(GlobalContext* globalCtx, Collider* collider) {
+    static Collider init = { NULL, NULL, NULL, NULL, 0, 0, 0, 0, 3, COLTYPE_INVALID };
+    *collider = init;
+    return 1;
+}
 
-s32 func_8005B6A0(GlobalContext* globalCtx, Collider* collision) {
+s32 Collider_DestroyBase(GlobalContext* globalCtx, Collider* collider) {
     return 1;
 }
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_collision_check/func_8005B6B0.s")
+s32 Collider_SetBase_Actor(GlobalContext* globalCtx, Collider* collider, ColliderInit_Actor* src) {
+    collider->actor = src->actor;
+    collider->atFlags = src->atFlags;
+    collider->acFlags = src->acFlags;
+    collider->maskA = src->maskA;
+    collider->maskB = 0x10;
+    collider->shape = src->shape;
+    return 1;
+}
 
-s32 func_8005B6EC(GlobalContext* globalCtx, Collider* collision, Actor* actor, ColliderBodyInfo* src) {
-    collision->actor = actor;
-    collision->unk_14 = src->unk_00;
-    collision->colliderFlags = src->colliderFlags;
-    collision->collideFlags = src->collideFlags;
-    collision->maskA = src->maskA;
-    collision->maskB = 0x10;
-    collision->type = src->maskB;
+// maskB = 0x10
+s32 Collider_SetBase_Set3(GlobalContext* globalCtx, Collider* collider, Actor* actor, ColliderInit_Set3* src) {
```

Comment:
same goes for all of the Set3's. What does this refer to? and is there anyway to make this fit the naming style better?

Reply:
just refers to the og set3 function the shapes have. rather than give the Collider_Set functions unique names for different arguments being passed in, it appears they just gave the functions names like Set, Set2, Set3, Set4, Set5, although there is no known 2 or 4 functions